crates/sui-types/src/storage/mod.rs

```diff
@@ -483,3 +483,10 @@ where
         self
     }
 }
+
+pub trait GetSharedLocks {
```

Comment:
So that
1. sui-transaction-checks crate doesn't need to depend on sui-core.
2. We could call into sui-transaction-checks in benchmarks using an in-memory store.